src/components/models/CartModel.ts

```diff
@@ -0,0 +1,232 @@
+import { Product, Promocode } from '../../types/interfaces';
+import AppController from '../app/app';
+import promos from '../../promocodes.json';
+
+export default class CartModel {
+    appController: AppController;
+    productsInCart: Product[];
+    totalSum: number;
+    productsQuantity: number;
+    promocodes: Promocode[];
+    enteredPromocodes: Promocode[];
+    activatedPromocodes: Promocode[];
+    totalSumWithDiscount: number;
+
+    constructor(controller: AppController) {
+        this.appController = controller;
+        this.promocodes = promos.promocodes;
+        this.totalSum = this.checkTotalSumInLocalStorage();
+        this.productsQuantity = this.checkProductQuantityInLocalStorage();
+
+        if (localStorage.getItem('cart')) {
```

Comment:
I suggest you to put "magic strings" into constants since you are using them in several different classes.

Reply:
fix it